cggmp21-keygen/src/threshold.rs

```diff
@@ -277,15 +277,16 @@ where
         .await
         .map_err(IoError::send_message)?;
 
-    for j in utils::iter_peers(i, n) {
+    let messages = utils::iter_peers(i, n).map(|j| {
         let message = MsgRound2Uni {
             sigma: sigmas[usize::from(j)],
         };
-        outgoings
-            .send(Outgoing::p2p(j, Msg::Round2Uni(message)))
-            .await
-            .map_err(IoError::send_message)?;
-    }
+        Outgoing::p2p(j, Msg::Round2Uni(message))
+    });
+    outgoings
+        .send_all(&mut futures::stream::iter(messages.map(Ok)))
+        .await
+        .map_err(IoError::send_message)?;
```

Comment:
Is it why you added `futures` dep, to do `futures::stream::iter`? You can use `futures-util` crate instead, i.e. [`futures_util::stream::iter`](https://docs.rs/futures-util/latest/futures_util/stream/fn.iter.html), `futures` crate just re-exports this function from `futures-util`

Reply:
Also, you could remove new (direct) dependency by using chain of [`outgoings.feed(msg).await`](https://docs.rs/futures-util/latest/futures_util/sink/trait.SinkExt.html#method.feed) inside of loop, followed by [`outgoings.flush().await`](https://docs.rs/futures-util/latest/futures_util/sink/trait.SinkExt.html#method.flush).

I would prefer this option, though using `futures_util::stream::iter` does have an advantage that we can't forget to call `.flush().await`.

Reply:
I see that in other places of the protocol, you are using `feed`/`flush` chain, so you can do that here as well and remove the dependency

Reply:
I use send_all where it doesn't change existing code too much, and feed+flush where send_all would. I think send_all is better, but I have a bias to not change existing lines too much